### PR TITLE
Limit response size

### PR DESCRIPTION
**Pull request by mrtcode**
Limits response size and allows JSDOM to filter `content-type`.

### CODE REVIEW

File: src/http.js
Comment (dstillman) [inline src/http.js]: Let's move this to a non-exported function outside of (and below) Zotero.HTTP.

File: src/http.js
Comment (dstillman) [inline src/http.js]: We should probably return either a 400 (as we do for an upstream 404) or a 501 (as we do for a document without a head) instead of a 500, and also return a specific error message. A 400 is probably better, since the request shouldn't be retried by the client anytime soon.

We can create a separate error type like Zotero.HTTP.StatusError and Zotero.HTTP.TimeoutError (above in this file) and throw that here, and then check for it in webSession.js.

**Review by dstillman — CHANGES_REQUESTED** [review]

File: src/http.js
Comment (mrtcode) [inline src/http.js:122]: For this we should also return 400, since we don't want to repeat translation too. It could be something like `InvalidResponse`. But maybe it's better to wait until we merge #59 because I suspect some changes for `content-type` filtering.

**Pull request merged**
